magma/config.py

```diff
@@ -22,3 +22,18 @@ def set_debug_mode(value=True):
 
 def get_debug_mode():
     return __DEBUG_MODE
+
+
+class _MagmaDebugSection:
```

Comment:
Perhaps this utility can go in a different file

Reply:
Nice, maybe in `testing`? One thing that I never got to was that this code should run even if the test fails (otherwise if the test fails, it leaves the debug mode on for other tests since it never hits the exit condition.

Reply:
I can move to `testing`.

I am seeing that `__exit__` runs even if the test fails (assuming it fails in a normal way). See:

```python
class _WithBlock:
    def __enter__(self):
        print ("Enter")

    def __exit__(self, typ, value, traceback):
        print ("Exit")

def test():
    with _WithBlock():
        assert False, "My assertion"

if __name__ == "__main__":
    test()
```